src/ZoneDifficulty.h

```diff
@@ -3,40 +3,64 @@
 
 #include "Player.h"
 #include "Config.h"
+#include "InstanceScript.h"
 #include "ScriptMgr.h"
 #include "ScriptedGossip.h"
 
-struct ZoneDifficultyData
+struct ZoneDifficultyNerfData
 {
     float HealingNerfPct;
     float AbsorbNerfPct;
     float SpellDamageBuffPct;
     float MeleeDamageBuffPct;
-    bool Enabled;
+    uint32 Enabled;
+    float HealingNerfPctHard;
+    float AbsorbNerfPctHard;
+    float SpellDamageBuffPctHard;
+    float MeleeDamageBuffPctHard;
+};
+
+struct ZoneDifficultyHardmodeInstData
+{
+    bool HardmodeOn;
+    bool HardmodePossible;
 };
 
 const int32 DUEL_INDEX = 0x7FFFFFFF;
 const int32 DUEL_AREA = 2402;       // Forbidding Sea (Wetlands)
+const uint32 NPC_TEXT_LEADER = 91301;
+const uint32 NPC_TEXT_OTHER = 91302;
+const __int8 MODE_NORMAL = 1;
+const __int8 MODE_HARD = 64;
```

Comment:
why not just int8?

Reply:
Copy Pasta and it worked.
Fixed!

Reply:
yea, however __intX is only defined on MSVC, so it would fail for any other compiles ;)